lib/circuits_sim/device/bmp280.ex

```diff
@@ -6,121 +6,129 @@ defmodule CircuitsSim.Device.BMP280 do
+          @bmp180_set_pressure_reading = value
+        ) do
+      registers =
+        state.registers
+        |> Map.put(reg, value)
+        |> Map.merge(Register.measurement_registers(:bmp180, :pressure))
 
-  defp put_raw_sample_registers(%{} = registers) do
-    [
-      # press_msb: 8 bytes from 0xF7
-      {0xF7, <<69>>},
-      {0xF8, <<89>>},
-      {0xF9, <<64>>},
-      {0xFA, <<130>>},
-      {0xFB, <<243>>},
-      {0xFC, <<0>>},
-      {0xFD, <<137>>},
-      {0xFE, <<109>>}
-    ]
-    |> Enum.into(registers)
-  end
+      %{state | registers: registers}
+    end
 
-  ## protocol implementation
+    def write_register(
+          %{sensor_type: :bmp180} = state,
+          @bmp180_reg_control_measurement = reg,
+          @bmp180_set_temperature_reading = value
+        ) do
+      registers =
+        state.registers
+        |> Map.put(reg, value)
+        |> Map.merge(Register.measurement_registers(:bmp180, :temperature))
 
-  defimpl SimpleI2CDevice do
-    @impl SimpleI2CDevice
-    def write_register(state, reg, value), do: put_in(state.registers[reg], <<value>>)
+      %{state | registers: registers}
+    end
+
+    def write_register(
+          %{sensor_type: :bmp280} = state,
+          @bmp280_reg_control_measurement = reg,
+          value
+        ) do
+      registers =
+        state.registers
+        |> Map.put(reg, value)
+        |> Map.merge(Register.measurement_registers(:bmp280))
+
+      %{state | registers: registers}
+    end
+
+    def write_register(
+          %{sensor_type: :bme280} = state,
+          @bme280_reg_control_measurement = reg,
+          value
+        ) do
+      registers =
+        state.registers
+        |> Map.put(reg, value)
+        |> Map.merge(Register.measurement_registers(:bme280))
+
+      %{state | registers: registers}
+    end
+
+    def write_register(state, reg, value) do
+      put_in(state.registers[reg], value)
+    end
 
     @impl SimpleI2CDevice
+    def read_register(
+          %{sensor_type: :bme680} = state,
+          @bme680_reg_status0
+        ) do
+      registers =
+        state.registers
+        |> Map.merge(Register.measurement_registers(:bme680))
+
+      new_data = 1
+      result = <<new_data::1, 0::7>>
+      {result, %{state | registers: registers}}
+    end
+
     def read_register(state, reg), do: {state.registers[reg], state}
 
     @impl SimpleI2CDevice
     def render(state) do
-      for {reg, data} <- state.registers do
-        [
-          "  ",
-          Tools.hex_byte(reg),
-          ": ",
-          for(<<b::1 <- data>>, do: to_string(b)),
-          "\n"
-        ]
-      end
+      rows =
+        for {addr_int, byte} <- Enum.sort_by(state.registers, fn {k, _} -> k end) do
+          addr_hex = Tools.hex_byte(addr_int)
+          bits = for(<<(b::1 <- <<byte>>)>>, do: to_string(b))
+          "#{addr_hex}: #{bits}\n"
+        end
+
+      ["Sensor type: #{state.sensor_type}\n" | rows]
```

Comment:
Printing all of the registers out doesn't seem that helpful compared to what's done with the other sensors. Since it's not easy to just print the temperature, humidity and pressure, how about just printing the sensor type.

```suggestion
      "Sensor type: #{state.sensor_type}\n"
```